scaladoc/src/dotty/tools/scaladoc/tasty/comments/MemberLookup.scala

```diff
@@ -236,12 +236,18 @@ object MemberLookup extends MemberLookup {
       // Scaladoc overloading support allows terminal * (and they're meaningless)
       val cleanStr = str.stripSuffix("*")
 
-      if cleanStr endsWith "$" then
+      cleanStr match {
+      case s"$name$$" if name.charAt(0).isUpper =>
```

Comment:
Hey! So I think these checks will have to be updated. It's not really safe to assume that the first letter here will be uppercased. While that may be somewhat common, there's no rule for that. So we can't rely on that here and down below.